onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -325,19 +333,40 @@ static bool MakeQDQNodeUnit(api::GraphRef& graph, const api::NodeRef& dq_node) {
     zp_value_info = graph.GetValueInfo(zp_input.value());
   }
 
-  // per-axis quantization if not a scalar (shape is empty for scalar).
+  // DQ uses per-axis quantization if scale is not a scalar. A scalar has an empty shape or a shape of [1].
   // note there could be an axis value as the onnx spec says that is ignored for per-tensor quantization,
   // so we have to check the shape.
-  auto update_dq_axis = scale_shape && !scale_shape->empty();
+  const bool scale_is_scalar = (scale_shape->empty() ||
+                                (scale_shape->size() == 1 && (*scale_shape)[0] == 1));
+  const bool update_dq_axis = !scale_is_scalar;
   int64_t axis = dq_node.GetAttributeIntDefault("axis", 1);
 
-  // TODO(adrianlizarraga): Also need to update axis if Unsqueeze inserts a 1 before the axis dim.
-  if (update_dq_axis && is_transpose) {
-    // update axis.
-    auto perm = GetPermAttrIfValid(next_node);
-    assert(perm.has_value());  // onnx shape inferencing checks that `perm` is valid
-    NormalizeAndValidateAxis(axis, scale_shape->size());
```

Comment:
`scale_shape->size()` is the wrong value to add to a negative axis. This PR instead uses dq_input[0]'s rank, and adds a unit test.